Providers/Scripts/2.4x-2.5x/Scripts/nxOMSKeyMgmt.py

```diff
@@ -0,0 +1,334 @@
+#!/usr/bin/env python
+#============================================================================
+# Copyright (C) Microsoft Corporation, All rights reserved.
+#============================================================================
+
+import os
+import sys
+import subprocess
+import imp
+import re
+import base64
+
+# class MSFT_nxOMSKeyMgmt:OMI_BaseResource
+# {
+#   [Key] string KeyContents;
+#   [Write,required] string KeySignature;
+#   [Write,ValueMap{"present", "absent"},Values{"present", "absent"}] string Ensure;
+# };
+
+protocol = imp.load_source('protocol', '../protocol.py')
+nxDSCLog = imp.load_source('nxDSCLog', '../nxDSCLog.py')
+
+LG = nxDSCLog.DSCLog
+
+dsc_keyring_path = '/etc/opt/omi/conf/omsconfig/keyring.gpg'
+signature_keyring_path = '/etc/opt/omi/conf/omsconfig/keymgmtring.gpg'
+gpg_bin = '/usr/bin/gpg'
```

Comment:
Whenever you run gpg, I think you need to make sure the "HOME" environment variable is set (and exported, so gpg can access it) to "/etc/opt/omi/conf/omsconfig"

By default, gpg stores some of its internal database stuff into the current user's home directory, but we modify that by setting HOME before we call gpg